File: backend/src/main/java/hu/bme/aut/workout_tracker_backend/data_layer/workout/Workout.java
package hu.bme.aut.workout_tracker_backend.data_layer.workout;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Workout {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO, generator="native")
    @GenericGenerator(name = "native")
    private Long id;
    private String name;
    private String userId;
    @ElementCollection
    private List<Long> exercises = new ArrayList<>();
}

File: workout-tracker-app-webservice/app/app/src/main/java/hu/bme/aut/workout_tracker/data/model_DEPRICATED/Workout.kt
package hu.bme.aut.workout_tracker.data.model_DEPRICATED

import java.util.UUID

data class Workout(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val user: String = "",
    val exercises: MutableList<String> = mutableListOf()
)